### PR TITLE
Update to fix #62377, #64045

### LINKED ISSUE #62377: [from customer] nios_txt_record module cannot handle multiple TXT records
##### SUMMARY

nios_txt_record module cannot handle multiple TXT records

##### ISSUE TYPE
- Bug Report

##### ANSIBLE VERSION

```paste below
2.8.2
```

##### CONFIGURATION

```paste below

```

##### STEPS TO REPRODUCE

Add the TXT record.  Change the "text" field value and add it again.  Instead of adding a second record with a different value, it replaces the first record.

I think it is an expected behavior with an idempotent way, but there is no way I can find to add multiple TXT records.

### PR DESCRIPTION
**Pull request by krisvasudevan**
##### SUMMARY

Fixes #62377, #64045

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

nios_a_record.py
nios_txt_record.py
api.py

##### ADDITIONAL INFORMATION

Added new parameters to A record and TXT record to handle modification of content in the records.
In case of A record, adding this new parameter enables modifying the IP address of a specific record when there are multiple records with the same name.
Similarly for TXT record, adding the new parameters enable modifying the text content of a specific TXT record when there are multiple records with the same name.

```paste below

```

### CODE REVIEW

**Review by Akasurde — CHANGES_REQUESTED** [review]
@krisvasudevan Thanks for the contribution. Could you please update the title of PR and description of PR to reflect what is being changed? Also, add a changelog entry [here](https://github.com/ansible/ansible/tree/devel/changelogs/fragments) so that we can backport this change?

Thanks.

**Comment by krisvasudevan** [comment]
My understanding from @justjais is that these changes cannot be backported since I have added new parameters. The bugs I fixed (#62377, #64045) needed new parameters to be added to the nios_a_record and nios_txt_record modules.

File: lib/ansible/modules/net_tools/nios/nios_a_record.py [hunk 1]
```diff
@@ -116,6 +116,17 @@
       password: admin
   connection: local
 
+- name: update an A record IP address
+  nios_a_record:
+    name: a.ansible.com
+    ipv4: {new_ipv4addr: 10.10.10.1, old_ipv4addr: 192.168.10.1}
```
Comment (justjais) [inline lib/ansible/modules/net_tools/nios/nios_a_record.py:123]: Plz comment the new param added in module doc (ref: [nios_host_record](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/net_tools/nios/nios_host_record.py#L31)).

**Review by justjais — CHANGES_REQUESTED** [review]

**Comment by justjais** [comment]
@krisvasudevan yes, this can't be backported as new params are introduced, and can you update the PR fixing the shippable failure. 
Also, can you update the PR description accordingly.

File: lib/ansible/modules/net_tools/nios/nios_a_record.py (hunk 1 above)
Comment (krisvasudevan) [reply lib/ansible/modules/net_tools/nios/nios_a_record.py:123]: Done for both nios_a_record and nios_txt_record

**Comment by krisvasudevan** [comment]
> @krisvasudevan yes, this can't be backported as new params are introduced, and can you update the PR fixing the shippable failure.
> Also, can you update the PR description accordingly.
Could you please give me some clarity on this?

File: lib/ansible/modules/net_tools/nios/nios_a_record.py [hunk 2]
```diff
@@ -27,7 +27,8 @@
   name:
     description:
       - Specifies the fully qualified hostname to add or remove from
-        the system
+        the system. User can also update the hostname as it is possible
+        to pass a dict containing I(new_name), I(old_name). See examples.
```
Comment (justjais) [inline lib/ansible/modules/net_tools/nios/nios_a_record.py]:
```suggestion
        to pass a dict containing I(new_ipv4addr), I(new_ipv4addr). See examples.
```

**Review by justjais — CHANGES_REQUESTED** [review]
Also, can u update the issue description with appropriate issue description as currently it mentions only the open issue IDs.

**Comment by krisvasudevan** [comment]
I've added information in the Additional Information section.

**Review by justjais — APPROVED** [review]

**Comment by sugitk** [comment]
@krisamundson 

Could you please update the status of this PR?  The customer is still waiting for the fix.
Thank you.

**Comment by danielmellado** [comment]
Thank you very much for your interest in Ansible. This plugin is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/community.general
Please re-submit this issue in the above repository.
If you have further questions please stop by IRC or the mailing list:

IRC: #ansible-network on irc.freenode.net
mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Pull request closed without merging**